src/lua/LuaPlayer.cpp

```diff
@@ -310,5 +313,32 @@ bool LuaPlayer::IsAttackable(unsigned char otherPlayerId)
 
 void LuaPlayer::SuggestPact(unsigned char otherPlayerId, const PactType pt, const unsigned duration)
 {
-    player.SuggestPact(otherPlayerId, pt, duration);
+    if (!player.isHuman())
+    {
+        AIPlayer* ai = GAMESERVER.GetAIPlayer(player.GetPlayerId());
+        AIInterface aii = ai->getAIInterface();
+        aii.SuggestPact(otherPlayerId, pt, duration);
+    }
 }
+
+void LuaPlayer::AcceptPact(const PactType pt, unsigned char otherPlayerId)
```

Comment:
This won't work. The GF passed to `AcceptPact` must be the GF where the pact was suggested. So you'd need to get it from the pact structs.
What is this method required for anyway? You have `EventSuggestPact` already.

Reply:
you are right, i removed this.